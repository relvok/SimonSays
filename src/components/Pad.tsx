import React from 'react';
import {View, TouchableOpacity} from 'react-native';
import {highlightPad} from 'utils/colors';
import {PadProps} from './types';
import styles from './styles';

const Pad = ({pad, onPadClickHandler, gameState}: PadProps) => {
  return (
    <View>
      <TouchableOpacity
        testID="pad-button"
        disabled={gameState.turn === 'simon'}
        activeOpacity={1}
        onPress={() => {
          onPadClickHandler(pad);
        }}
        style={[
          styles.pad,
          {
            backgroundColor: pad.isActive ? highlightPad(pad) : pad.value,
            transform: [{rotate: pad.percent}],
          },
        ]}></TouchableOpacity>
    </View>
  );
};

export default Pad;
