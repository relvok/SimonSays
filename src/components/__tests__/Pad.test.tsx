import React from 'react';
import renderer from 'react-test-renderer';
import {fireEvent, act} from '@testing-library/react-native';
import Pad from 'components/Pad';

describe('StartButton component', () => {
  const onPadClickHandler = jest.fn();
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Pad
          pad={{name: '', value: '', isActive: true, percent: ''}}
          onPadClickHandler={onPadClickHandler}
          padsDisabled={false}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls onPadClickHandler when button is pressed', () => {
    const component = renderer.create(
      <Pad
        pad={{name: '', value: '', isActive: true, percent: ''}}
        onPadClickHandler={onPadClickHandler}
        padsDisabled={false}
      />,
    );
    const instance = component.root;
    const button = instance.findByProps({
      testID: 'pad-button',
    });
    act(() => {
      fireEvent.press(button);
    });
    expect(onPadClickHandler).toHaveBeenCalled();
  });
});
