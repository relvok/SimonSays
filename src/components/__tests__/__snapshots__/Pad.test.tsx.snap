// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartButton component renders correctly 1`] = `
<View>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": undefined,
        "borderBottomLeftRadius": 0,
        "borderBottomRightRadius": 10,
        "borderColor": "white",
        "borderTopLeftRadius": 130,
        "borderTopRightRadius": 0,
        "borderWidth": 0,
        "height": 130,
        "margin": 5,
        "opacity": 1,
        "transform": Array [
          Object {
            "rotate": "",
          },
        ],
        "width": 130,
      }
    }
    testID="pad-button"
  />
</View>
`;
